R/gt_format.R

```diff
@@ -0,0 +1,165 @@
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#' This is the function to format the bounds summary table into gt style.
+#'
+#' @param output an object returned by \code{summary_bound}
+#' @param method he method used to generate x. It should be one of \code{c("AHR", "WLR", "COMBO")}
+#' @param title a string to specify the title of the gt table
+#' @param subtitle a string to specify the subtitle of the gt table
+#' @param colname_spanner a string to specify the spanner of the gt table
+#' @param colname_spannersub a vector of strings to specify the spanner details of the gt table
+#' @param footnote a list containing \code{content}, \code{location}, and \code{attr}. 
+#'                 the \code{content} is a vector of string to specify the footnote text;
+#'                 the \code{location} is a vector of string to specify the locations to put the superscript of the footnote index;
+#'                 the \code{attr} is a vector of string to specify the attributes of the footnotes, e.g., c("colname", "title", "subtitle", "analysis", "spanner");
+#'                 users can use the functions in the \code{gt} package to custom themselves.
+#'
+#' @return
+#' @export
+#'
+#' @examples 
+#' x_combo <- gsdmvn::gs_design_combo2(
+#'   ratio = 1, 
+#'   alpha = 0.025, 
+#'   beta = 0.2, 
+#'   enrollRates = tibble::tibble(Stratum = "All", duration = 12, rate = 500/12),
+#'   failRates = tibble::tibble(Stratum = "All", duration = c(4, 100), 
+#'                              failRate = log(2) / 15, hr = c(1, .6), dropoutRate = .001), 
+#'   fh_test = fh_test,
+#'   upper = gsdmvn::gs_spending_combo,
+#'   upar = list(sf = gsDesign::sfLDOF, total_spend = 0.025),
+#'   lower = gsdmvn::gs_spending_combo,
+#'   lpar = list(sf = gsDesign::sfLDOF, total_spend = 0.2))
+#'   summary_bound(x_combo2, method = "COMBO") %>% 
+#' gt_format(method = "COMBO", 
+#'           title = "Summary of the Crossing Probability",
+#'           subtitle = "by Using gs_design_wlr",
+#'           colname_spanner = "Cumulative boundary crossing probability",
+#'           colname_spannersub = c("Alternate hypothesis", "Null hypothesis"),
+#'           footnote = list(content = c("gs_design_wlr is a function in gsdmvn.",
+#'                                       "EF is the events fraction, i.e., Events/N."),
+#'                           location = c(NA, NA),
+#'                           attr = c("subtitle", "analysis")))
+#'                           
+gt_format <- function(
```

Comment:
Does it make sense to name this function `as_gt()` which essentially converts the `summary_bound` object to a `gt` object? This way you can extend the output type to multiple table frameworks like [what gtsummary did](https://www.danieldsjoberg.com/gtsummary/reference/index.html#output-types).